src/components/ProductsPage/ProductsPage.js

```diff
@@ -1,20 +1,46 @@
-import { React, useState } from "react";
+import { React, useState, useEffect } from "react";
 import Sidebar from "../Sidebar/Sidebar.component";
 import Products from "../Products/Products.component";
-import products from "../../mocks/products.json";
-import CatList from "../../mocks/product-categories.json";
+import { useCategories } from "../../utils/hooks/useCategories";
+import { useProducts } from "../../utils/hooks/useProducts";
 
 function ProductsPage() {
   const [categories, setCategories] = useState([]);
+  const { data: catList, isLoading: catIsLoading } = useCategories();
+  const { data: products, isLoading: productsIsLoading } = useProducts();
+
+  const urlCategory = new URLSearchParams(window.location.search).get(
```

Comment:
i don't really get this. Doing just a very small test I don't seem to find a real use as this value is always null

Reply:
I'm gonna leave a comment at the top level of the pr to not attach it to one component regarding the fetching of filtered products of one or more categories

Reply:
Yeah this was a little bit rushed lol, but the purpose of this is so you can click on a category on the homepage and it leads you to the productsPage with a category set from the URL. I thought about also modifying the URL based on which categories are selected but I did not have time for that.